classes/class-wpcom-liveblog-rest-api.php

```diff
@@ -0,0 +1,496 @@
+<?php
+
+/**
+ * Class WPCOM_Liveblog_Rest_Api
+ *
+ * This class integrates with the REST API framework added in WordPress 4.4
+ * It registers endpoints matching the legacy functionality in the WPCOM_Liveblog ajax methods
+ *
+ */
+
+class WPCOM_Liveblog_Rest_Api {
+
+	private static $api_version;
+	private static $api_namespace;
+
+	public static $endpoint_base;
+	public static $endpoint_get_entries_by_date;
+	public static $endpoint_crud;
+
+	/**
+	 * Load everything the class needs
+	 */
+	public static function load() {
+
+		self::$api_version   = '1';
+		self::$api_namespace = 'liveblog/v' . self::$api_version;
+		if ( get_option( 'permalink_structure' ) ) {
+			// Pretty permalinks enabled 
+			self::$endpoint_base = '/wp-json/' . self::$api_namespace . '/';
```

Comment:
Can we get the API base programatically instead of hard-coding it? (I believe the `wp-json` part can change.)